### ISSUE TITLE
Unified Memory Support in ROCm

### ISSUE THREAD

**Issue by misos1** [hipSYCL/compiler-support#3]
> [1] HIP/ROCm implements unified memory using slow device accessible host memory. This means that hipSYCL's call to hipMallocManaged cannot produce efficient shared allocations.

Is this still true? There is some change in version 4.5:

https://github.com/RadeonOpenCompute/ROCm#unified-memory-support-in-rocm

**Comment by illuhad** [comment]
You are right, things are changing in ROCm. As hipSYCL just calls `hipMallocManaged()`, those changes directly translate to hipSYCL as well. However, from what I have seen and heard, performance of `hipMallocManaged()` even with page migration is still very, very bad.
